core/settings.py

```diff
@@ -38,8 +39,54 @@
     'rest_framework.authtoken',
     'djoser',
     'corsheaders',  # Added for CORS support
+    
+    # OAuth
+    'allauth',
+    'allauth.account',
+    'allauth.socialaccount',
+    'allauth.socialaccount.providers.google',
+    'allauth.socialaccount.providers.github',
 ]
 
+SITE_ID = 1
+
+AUTHENTICATION_BACKENDS = [
+    'django.contrib.auth.backends.ModelBackend',
+
+    'allauth.account.auth_backends.AuthenticationBackend',
+]
+
+
+SOCIALACCOUNT_PROVIDERS = {
+    'google': {
+        'APP': {
+            'client_id': config('GOOGLE_CLIENT_ID'),
+            'secret': config('GOOGLE_CLIENT_SECRET'),
+            'key': '',
+        },
+        'SCOPE': ['profile', 'email'], 
+        'AUTH_PARAMS': {
+            'access_type': 'online',
+            'prompt': 'select_account', 
+        },
+        'FETCH_USERINFO': True,  
+    },
+
+    'github': {
+        'APP': {
+            'client_id': config('GITHUB_CLIENT_ID'),
+            'secret': config('GITHUB_CLIENT_SECRET'),
+            'key': '',
+        },
+        'SCOPE': ['user:email'],
+    }
+}
```

Comment:
Consider handling missing configuration values gracefully, e.g., log a warning if a key is not set.
For Google, consider adding 'offline' to `access_type` if you plan to use refresh tokens.